src/kafka-file/codeLensProvider.ts

```diff
@@ -112,27 +118,63 @@ export class KafkaFileCodeLensProvider implements vscode.CodeLensProvider, vscod
     }
 
     private createProducerLens(lineRange: vscode.Range, range: vscode.Range, document: vscode.TextDocument, clusterName: string | undefined): vscode.CodeLens[] {
-        const produceRecordCommand = this.createProduceRecordCommand(document, range);
+        const produceRecordCommand = this.createProduceRecordCommand(document, range, this.clusterSettings.selected?.id);
         const lenses: vscode.CodeLens[] = [];
         if (clusterName) {
-            // Add Produce lenses
-            lenses.push(new vscode.CodeLens(lineRange, {
-                title: "$(run) Produce record",
-                command: ProduceRecordCommandHandler.commandId,
-                arguments: [produceRecordCommand, 1]
-            }));
-            lenses.push(new vscode.CodeLens(lineRange, {
-                title: "$(run-all) Produce record x 10",
-                command: ProduceRecordCommandHandler.commandId,
-                arguments: [produceRecordCommand, 10]
-            }));
+            const producerUri = createProducerUri(produceRecordCommand);
+            const producer = this.producerCollection.get(producerUri);
+            const producerState = producer ? producer.state : ProducerLaunchState.none;
+            switch (producerState) {
+                case ProducerLaunchState.connecting:
+                case ProducerLaunchState.disconnecting:
+                case ProducerLaunchState.sending:
+                case ProducerLaunchState.sent:
```

Comment:
if state is sent, you don't want the spinner

Reply:
fixed